keps/sig-api-machinery/2885-server-side-unknown-field-validation/README.md

```diff
@@ -0,0 +1,963 @@
+
+This section must be completed when targeting beta to a release.
+
+<!--
+###### How can an operator determine if the feature is in use by workloads?
+
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+###### How can someone using this feature know that it is working for their instance?
+
+The easiest way to know that the feature is enabled and working is to query the
+API server with the query param set and passing an object with deliberately
+invalid, unknown fields. An error from the server will indicate that the feature
+is enabled as expected.
+
+Alternatively, and end-user can cross-reference the results of client-side
+validation (i.e. `kubectl --validate=true`) with the results of server-side. If
+client-side validation indicates an error but server-side does not, then the
+feature is disabled.
+
+<!--
+
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+
+- [ ] Events
+  - Event Reason: 
+- [ ] API .status
+  - Condition name: 
+  - Other field: 
+- [ ] Other (treat as last resort)
+  - Details:
+-->
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+Users should expect to see an increase in request latency and memory usage
+(~20-25% increase) for requests that desire server side schema validation.
+
+Cluster operators can also use cpu usage monitoring to determine whether
+increased usage of server-side schema validation dramatically increases CPU usage after feature enablement.
+
+<!--
+This is your opportunity to define what "normal" quality of service looks like
+for a feature.
+
+It's impossible to provide comprehensive guidance, but at the very
+high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99.9% of /health requests per day finish with 200 code
+
+These goals will help you determine what you need to measure (SLIs) in the next
+question.
+
+-->
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+Pick one more of these and delete the rest.
+
+- [x] Metrics
+  - Metric name: CPU Usage
+  - Components exposing the metric: kube-apiserver
+- [x] Metrics
+  - Metric name: Memory Consumption
+  - Components exposing the metric: kube-apiserver
+- [x] Metrics
+  - Metric name: Request Latency
```

Comment:
Let's provide the exact metric name:
apiserver_request_duration_seconds

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go